### ISSUE TITLE
Setup SSO with ory.sh

### ISSUE THREAD

**Issue by kelson42** [kiwix/overview#90]
First service to be migrated to (via Saml or OIDC) should be Bitwarden

**Comment by rgaudin** [comment]
What does that mean? We have not BW service AFAIK

**Comment by kelson42** [comment]
> What does that mean? We have not BW service AFAIK

We do (for the backup service) and we will have to relaunch it (maybe with an onprem instance), ticket to follow.

But ory.sh should be ready get OIDC/SAML requests first.

**Comment by rgaudin** [comment]
> We do (for the backup service) and we will have to relaunch it (maybe with an onprem instance), ticket to follow.

Not our service. We use the SaaS version with one (actually two) backup-dedicated Free accounts.
Maybe we want to have shared BW secrets for the team and would use SSO to connect to BW's SaaS or we could setup our own BW and use SSO to access it in the same manner.

Either way it's not a _service to migrate_ (from our POV) ory.sh must be setup indeed and tested.
FYI BW supports both SAML and OIDC.

Should you want to extend borg-backup's behavior, a ticket there would be needed.

**Comment by rgaudin** [comment]
- Centralized/unified authentication to services we use (3rd parties and our own)
  - MFA requirement/policy
  - Configure OpenID Connect

--

Our services to accept any OpenID Connect
- ory.sh should be able to manage identifies
- Document first service deployment/share code for easy duplication in other services

--

- Decide on one third-party service to validate solution with.
- Decide on one of our service to validate solution with.

--

- Roles at Ory level (org-level roles: function-related). ex: `content-editor`
- Permissions must be handled in apps
- Ex for cms: admin on a catalog or not

**Comment by benoit74** [comment]
Is anyone of the team aware of any third-party service (e.g. BW) where we could plug our own identity provider at no cost? 

<img width="1423" height="872" alt="Image" src="https://github.com/user-attachments/assets/161bb1ff-3d3a-4848-9fca-4b54f6cdf784" />

The only possibility with Bitwarden teams is for instance to provision users dynamically (either with their "Directory connector" or with SCIM). We need Bitwarden Enterprise to connect SSO.

Same on Github:

<img width="1271" height="856" alt="Image" src="https://github.com/user-attachments/assets/50d0db71-91aa-4ecc-a288-985a9349037d" />

I did not checked all other services, but I doubt we can find many where we can do it with our current tier. Not speaking about the fact that we tend to reduce the number of accounts on these third-party (or even use a single shared account).

Do we really want to investigate SSO for third-parties?

Even reducing the scope to only provisioning is going to be a pain: most do not support SCIM except in "Enterprise" tier AFAIK, and SCIM is anyway a pain to setup (you need to provision / unprovision test users multiple times to check behavior).

**Comment by kelson42** [comment]
SSO is first for our solution and then for third party if we can afford it (yes, because this is cost related).

**Comment by benoit74** [comment]
Deployed on farm.openzim.org and farm.zimit.kiwix.org, just waiting for a small fix to make 2FA mandatory.

Ready to be deployed on CMS once upgrade branch will be moved to production.

**Comment by benoit74** [comment]
And mostly everything is documented at https://github.com/kiwix/overview/wiki/Ory.sh-SSO

**Comment by rgaudin** [comment]
Considering it's already in production on Zimfarm and CMS, this meta ticket has is not required anymore.

**Comment by benoit74** [comment]
Indeed, I was waiting for 2FA to be mandatory on both Zimfarm and CMS to close the issue. It is now the case.
